src/main/java/net/greatstart/controllers/InvestmentController.java

```diff
@@ -1,31 +1,115 @@
+import java.security.Principal;
+import java.time.LocalDateTime;
 import java.util.List;
 
 @Controller
 public class InvestmentController {
+    private static final String PROJECT_PAGE = "project/";
+    private static final String INVESTMENTS_VIEW = "investment/investments";
+    private static final String INVESTMENTS_PAGE = "investment";
+    private static final String EDIT_USER_PROFILE = "user/EditUserPage";
+    private static final String REDIRECT = "redirect:/";
 
     private InvestmentService investmentService;
+    private ProjectService projectService;
+    private UserService userService;
 
     @Autowired
-    public InvestmentController(InvestmentService investmentService) {
+    public InvestmentController(InvestmentService investmentService,
+                                ProjectService projectService, UserService userService) {
         this.investmentService = investmentService;
+        this.projectService = projectService;
+        this.userService = userService;
     }
 
-    public Investment saveInvestment(Investment investment) {
-        return investmentService.saveInvestment(investment);
+    @PreAuthorize("isAuthenticated()")
+    @GetMapping("/project/{id}/addInvestment")
+    public ModelAndView getAddInvestmentForm(@PathVariable long id) {
+        Project project = projectService.getProjectById(id);
+        ModelAndView model = new ModelAndView("investment/add_investment");
+        model.addObject(project);
+        model.addObject("investedSum", getInvestedSumInProject(id));
+        return model;
     }
 
-    public Investment getInvestmentById(long id) {
-        return investmentService.getnvestmentById(id);
+    @PostMapping("/project/{id}/addInvestment")
+    public ModelAndView addInvestment(@PathVariable long id,
+                                      @RequestParam BigDecimal sum,
+                                      Principal principal) {
+        Project project = projectService.getProjectById(id);
+        User investor = userService.getUserByEmail(principal.getName());
+        String message = InvestmentValidationService.validate(sum, project);
+        if (investor.getContact().getPhoneNumber() == null
+                || investor.getContact().getPhoneNumber().isEmpty()) {
+            ModelAndView model = new ModelAndView(REDIRECT + EDIT_USER_PROFILE);
+            model.addObject("message", "Please enter your phone number!");
+            return model;
+        }
+        if (message != null) {
+            ModelAndView model = new ModelAndView("investment/add_investment");
+            model.addObject("message", message);
+            model.addObject(project);
+            model.addObject("investedSum", getInvestedSumInProject(id));
+            return model;
+        }
+        investmentService.saveInvestment(new Investment(LocalDateTime.now(),
+                project, investor, sum, false, false));
+
+        return new ModelAndView(REDIRECT + PROJECT_PAGE + id);
```

Comment:
Create empty model (ModelAndView). Fill it depending on condition and return at the end instead of returning 3 different models.